src/home_robot/home_robot/mapping/voxel.py

```diff
@@ -25,37 +30,177 @@ def combine_point_clouds(
+    xx, yy = np.meshgrid(x, y, indexing="ij")
+
+    # Compute the distance transform
+    distance_map = np.sqrt((xx - size // 2) ** 2 + (yy - size // 2) ** 2)
+
+    # Create the disk by thresholding the distance transform
+    disk = distance_map <= radius
+
+    return disk
 
 
 class SparseVoxelMap(object):
     """Create a voxel map object which captures 3d information."""
 
-    def __init__(self, resolution=0.01, feature_dim=3):
+    def __init__(
+        self,
+        resolution=0.01,
+        feature_dim=3,
+        grid_size: Tuple[int, int] = None,
+        grid_resolution: float = 0.05,
+        obs_min_height: float = 0.1,
+        obs_max_height: float = 1.8,
+        obs_min_density: float = 5,
+        add_local_radius_points: bool = True,
+        local_radius: float = 0.25,
+        cast_to_center: bool = False,
+        min_depth: float = 0.1,
+        max_depth: float = 4.0,
+    ):
         self.resolution = resolution
         self.feature_dim = feature_dim
-        self.xyz = None
-        self.feats = None
         self.observations = []
+        self.obs_min_height = obs_min_height
+        self.obs_max_height = obs_max_height
+        self.obs_min_density = obs_min_density
+        self.grid_resolution = grid_resolution
+        self._seq = 0
+        self._2d_last_updated = -1
+        self.min_depth = min_depth
+        self.max_depth = max_depth
 
-    def add(self, camera_pose: np.ndarray, xyz: np.ndarray, feats: np.ndarray, **info):
-        """Add this to our history of observations. Also update the current running map."""
+        # Create disk for mapping explored areas near the robot - since camera can't always see it
+        self._disk_size = np.ceil(1.0 / self.grid_resolution)
+        self._visited_disk = create_disk(
+            1.0 / self.grid_resolution, (2 * self._disk_size) + 1
+        )
+
+        # Add points with local_radius to the voxel map at (0,0,0) unless we receive lidar points
+        self.add_local_radius_points = add_local_radius_points
+        self.local_radius = local_radius
+        # TODO:
+        # Cast lines to the center from nearest points - this is to create explored re
+        if cast_to_center is True:
+            raise NotImplementedError("cast to center not implemented")
+
+        if grid_size is not None:
+            self.grid_size = [grid_size[0], grid_size[1]]
+        else:
+            self.grid_size = DEFAULT_GRID_SIZE
+        # Track the center of the grid - (0, 0) in our coordinate system
+        # We then just need to update everything when we want to track obstacles
+        self.grid_origin = np.array(self.grid_size + [0]) // 2
+        # Create map here
+        self.reset_cache()
+
+    def reset_cache(self):
+        """Clear some tracked things"""
+
+        # Stores points in 2d coords where robot has been
+        self._visited = np.zeros(self.grid_size)
+
+        # Store instances detected (all of them for now)
+        self._instance_views = []
```

Comment:
We should create a single `InstanceMemory` object instead. This should handle the addition of temporary local instance views (`process_instances_for_env`), and mapping local views to global instances (`update_instance_id`).

Reply:
I think the reason i didnt do this is because i dont have a global semantic map?

Reply:
added code to use this; now we need to gfix it